library/std/src/os/mod.rs

```diff
@@ -78,18 +78,18 @@ pub mod linux;
 #[cfg(not(all(
     doc,
     any(
-        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_arch = "wasm32", not(target_os = "wasi"), not(target_os = "wasi-preview2")),
         all(target_vendor = "fortanix", target_env = "sgx")
     )
 )))]
-#[cfg(any(target_os = "wasi", doc))]
+#[cfg(any(target_os = "wasi", target_os = "wasi-preview2", doc))]
 pub mod wasi;
```

Comment:
I'd recommend actually removing this for `wasi-preview2` and we can add it back in incremental as-necessary once we figure out how the preview2<->libc<->libstd bridge all works (unless @dicej has code using this and needs it working in the near future)